vectorial_model.py

```diff
@@ -0,0 +1,141 @@
+import math
+from collections import Counter
+from collections import OrderedDict
+
+from collection_processing import article_tokenize_other, lemmatize, remove_stop_words
+
+
+def remove_non_index_term(query, inverted_index):
+    """"Remove tokens that are not in index from query."""
+    query_filtered = []
+    for token in query:
+        if token in inverted_index:
+            query_filtered.append(token)
+    return query_filtered
+
+
+def pre_processed_query(query, inverted_index, stop_words):
+    """
+    Process raw query for vectorial vectorial query.
+
+    Tokenize, filter stop words and lemmatize query.
+    :param query: (string) query
+    :param inverted_index:
+    :return:
+    """
+    tokenized_query = article_tokenize_other(query)
+    filtered_query = remove_non_index_term(tokenized_query, inverted_index)
+    filtered_query = remove_stop_words(filtered_query, stop_words)
+    normalized_query = lemmatize(filtered_query)
+    return normalized_query
+
+
+def get_term_frequency(term, doc_id, index_frequency):
+    """Get the frequency of a term in a document of the collection."""
+    return index_frequency[term][doc_id]
+
+
+def get_logarithmic_term_frequency(term, doc_id, index_frequency):
+    """
+    Get logarithmic frequency of term in a document of the collection.
+
+    Equal to 0 if the term is not in the document, otherwise 1 + log(term_frequency).
+    """
+    tf = get_term_frequency(term, doc_id, index_frequency)
+    if tf > 0:
+        return 1 + math.log(tf)
+    else:
+        return 0
+
+
+def get_normalized_term_frequency(term, doc_id, index_frequency, stats_collection):
+    """Get normalized term frequency"""
+    term_frequency = get_term_frequency(term, doc_id, index_frequency)
+    normalized_term_frequency = 0.5 + 0.5 * (term_frequency / stats_collection[doc_id]["max_frequency"])
+    return normalized_term_frequency
+
+
+def get_normalized_logarithmic_term_frequency(term, doc_id, index_frequency, stats_collection):
+    """Get normalized logarithmic term frequency"""
+    term_frequency = get_term_frequency(term, doc_id, index_frequency)
+    normalized_logarithmic_term_frequency = (1 + math.log(term_frequency)) / (1 + math.log(
+        stats_collection[doc_id]["average_frequency"]))
+    return normalized_logarithmic_term_frequency
+
+
+def get_idf(term, index_frequency, nb_doc):
+    """Gets idf of term in collection"""
+    return math.log(nb_doc / len(index_frequency[term].keys()))
+
+
+def get_stats_document(document):
+    """
+    Get statistics on the document.
+
+    Calculates the frequency of most common term, the number of unique terms, and the average frequency of terms.
+    :param document: Document to get statistics on
+    :return: (dict) stats
+    """
+    counter = Counter()
+    for term in document:
+        counter.update([term])
```

Comment:
Why not this ?

```suggestion
    counter = Counter(document)
```

I'm 90% it would work.
https://docs.python.org/3/library/collections.html#collections.Counter

Reply:
Yep changed this, I had just copied what was in the lab.